sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -78,9 +78,42 @@ case class ProjectExec(projectList: Seq[NamedExpression], child: SparkPlan)
     }
   }
 
-  override def outputOrdering: Seq[SortOrder] = child.outputOrdering
```

Comment:
Take the plan in PR description as an example. In the `Project`,

    Project [named_struct(_1, _1#83, _2, _2#84) AS _1#105]

Its output is ` _1#105`. Its `outputPartitioning` is  `_1#83`. But the parent operator of the `Project` requires partitioning on ` _1#105._1`.  Since `_1#83.semanticEquals(_1#105._1)` is false, additional ShuffleExchange will be added. 

`outputOrdering` is the same issue.